File: src/API/Microsservices/Marketplace/Sonorus.Marketplace.Application/Commands/UpdateProduct/UpdateProductInputModel.cs
using Microsoft.AspNetCore.Http;
using Sonorus.Marketplace.Core.Enums;

namespace Sonorus.Marketplace.Application.Commands.UpdateProduct;

public class UpdateProductInputModel {
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public ConditionType Condition { get; set; }
    public ICollection<IFormFile> NewMedias { get; set; } = [];
    public ICollection<long> MediasToRemove { get; set; } = [];
}

File: src/app/lib/src/dtos/input_models/update_product_input_model.dart
import "package:dio/dio.dart";
import "package:image_picker/image_picker.dart";

import "package:sonorus/src/domain/enums/condition_type.dart";

class UpdateProductInputModel {
  final String name;
  final double price;
  final String? description;
  final ConditionType condition;
  final List<XFile> newMedias;
  final List<int> mediasToRemove;

  UpdateProductInputModel({
    required this.name,
    required this.price,
    this.description,
    required this.condition,
    required this.newMedias,
    required this.mediasToRemove
  });

  FormData toFormData() {
    final FormData formData = FormData.fromMap({
      "name": this.name,
      "price": this.price.toString().replaceAll(".", ","),
      "description": this.description,
      "condition": this.condition.id,
      "mediasToRemove": mediasToRemove
    });
    formData.files.addAll(newMedias.map((media) => MapEntry("newMedias", MultipartFile.fromFileSync(media.path))));
    return formData;
  }
}